site/_posts/2017-10-24-api-contributors-guide.md

```diff
@@ -0,0 +1,250 @@
+        :options: 
+        - :collection
+        :verbs: *gpd
+        :klass: Coffee
+        :collection_actions:
+            :get: 
+            - :name: read
+              :identifier: coffee_show_list 
+            :post:
+            - :name: create
+              :identifier: coffee_new 
+            - :name: edit
+              :identifier: coffee_edit 
+            - :name: delete
+              :identifier: coffee_delete
+        :resource_actions:
+            :get:
+            - :name: read
+              :identifier: coffee_show 
+            :post:
+            - :name: edit
+              :identifier: coffee_edit 
+            - :name: delete
+              :identifier: coffee_delete
+            :delete:
+            - :name: delete
+              :identifier: coffee_delete
+```
+
+
+Here is a breakdown of what some of these elements mean:
+* **: coffee:**
+    * Defines the route /api/coffee
+    * Tells the API that the controller will be `CoffeeController`
+* **:options:**
+    * Gives more details about the collection and whether it will act as a collection, subcollection
+* **:verbs:**
+    * Specifies the [verb set](https://github.com/ManageIQ/manageiq-api/blob/master/config/api.yml#L18-L36) that can be used on the collection
+* **:klass:** 
+    * Specifies the base class name of the resources under this collection 
+* **:name:**
+    * Name of the action to be performed 
+* **:identifier:**
+    * Name of the corresponding [product feature](https://github.com/ManageIQ/manageiq/blob/master/db/fixtures/miq_product_features.yml), which is used to control permissions to resources
+* **:collection_actions:** vs **:resource_actions:**
+    * collection actions correspond to those against the collection `/api/coffee` and resource actions are those corresponding to a single resource `/api/coffee/:c_id`
+ 
+Through the above definition, the following endpoints and actions were created:
+```
+GET /api/coffee
+GET /api/coffee/:c_id
+
+POST /api/coffee
+POST /api/coffee { "action": "edit" }
+POST /api/coffee { "action": "delete" }
+POST /api/coffee/:c_id { "action": "edit" } 
+POST /api/coffee/:c_id { "action": "delete" } 
+
+DELETE /api/coffee/:c_id
+```
+### **Where are the methods?**
+Now that the routes exist, let's take care of the methods. Once the new collection has been added to the `api.yml`, the corresponding controller needs to be [created here](https://github.com/ManageIQ/manageiq-api/tree/master/app/controllers/api). The controller will look something like:
+```ruby
+module Api
+    class CoffeeController < BaseController 
+    end 
+end 
+```
+Because the `CoffeeController` inherits from the `BaseController`, a lot of what has been defined in the `api.yml` will already work :sparkles:. The [Generic Module](https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/base_controller/generic.rb) holds the basic CRUD methods that will be called if they are not overridden in the `CoffeeController`.
```

Comment:
I think to clarify here you may want to say "action methods" i.e. the ones that are specifically `#show`, `#index`, etc..  Also, those [now live in base_controller.rb](https://github.com/ManageIQ/manageiq-api/pull/95) 😀